### ISSUE TITLE
Accounts per Project (Suitecloud for vscode)

### ISSUE THREAD

**Issue by mtoniato** [oracle/netsuite-suitecloud-sdk#542]
As far as I know the accounts that are configured in Visual Studio Code with "Suitecloud: Set Up Accounts" are global and not project-based.

In my case I am working on many Netsuite platforms. I created separated projects but it is very dangerous to have shared accounts: accidentally I could deploy the file on the wrong Netsuite environment.

**Comment by alii-gator** [comment]
This is by design and default authids are set per project.
User can set default account in project.json which doesn't need to be changed whenever you deploy.
What's your suggestion or desired flow?

**Comment by mtoniato** [comment]
Thanks Mohammed for your response.
The focus of my issue is not related to the DEFAULT auth ids but rather to the LIST of the available auth ids.
Tipically for each project I'll have two auth ids (one for production, one for sandbox) and I'll change the default depending on the needs.

What in my opinion is very risky is the possibility to choose the default id among all the ids, including those accounts that are not actually related with the current project.

The expected behavior is a list that is filtered by project. I think that this could be achieved by not saving the accounts globally but in a project file instead.

How do you feel about it?

Thanks in advance.

**Comment by mtoniato** [comment]
Thanks Mohammed for your response.
The focus of my issue does not concern the default auth ids but the list of the available auth ids.
Tipically I'll have one project and two auth ids (one for production, one for sandbox) and change the default depending on the needs.

What in my opinion is very risky is to choose the default id among all the ids, including those accounts that are not related with current project.

The expected behavior is a restricted list by project. I think that this could be achieved by not saving the accounts globally but in a project file.

Thanks in advance,
Manuel



[ALLYS]<http://allys.it/>

Manuel TONIATO
SENIOR CONSULTANT
M +39 327 775 3631 | ***@***.******@***.***> | LinkedIn<https://www.linkedin.com/in/manuel-toniato-61b07a45/>
allys.it<http://allys.it/> | FOLLOW US ON LinkedIn<https://www.linkedin.com/company/allysrl/>

[Powered by Altea Federation]<http://www.alteafederation.it/>


________________________________
Da: Syed Mohammad Ali ***@***.***>
Inviato: mercoledì 16 marzo 2022 23:23
A: oracle/netsuite-suitecloud-sdk ***@***.***>
Cc: Manuel Toniato ***@***.***>; Author ***@***.***>
Oggetto: Re: [oracle/netsuite-suitecloud-sdk] Accounts per Project (Suitecloud for vscode) (Issue #542)


This is by design and default authids are set per project.
User can set default account in project.json which doesn't need to be changed whenever you deploy.
What's your suggestion or desired flow?

—
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Foracle%2Fnetsuite-suitecloud-sdk%2Fissues%2F542%23issuecomment-1069696372&data=04%7C01%7Cmtoniato%40alteanet.it%7Ca3c4262c9bb94c3c65bd08da079b956c%7Ca248a8b3ddc049da9a7070c77203e1d7%7C1%7C0%7C637830662050273821%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=oF5FNjXXfUocs47H3%2F%2FX2ANMLZx4fFrwqyxISnwGqMg%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAXWJLMGNYUMWC6FEIL2HA7DVAJNNVANCNFSM5OX542PA&data=04%7C01%7Cmtoniato%40alteanet.it%7Ca3c4262c9bb94c3c65bd08da079b956c%7Ca248a8b3ddc049da9a7070c77203e1d7%7C1%7C0%7C637830662050273821%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=u7dLTO5ef%2BqQ7Ho0IzEpMZrU1kK0h8QnnfrO0%2BSoG8k%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cmtoniato%40alteanet.it%7Ca3c4262c9bb94c3c65bd08da079b956c%7Ca248a8b3ddc049da9a7070c77203e1d7%7C1%7C0%7C637830662050273821%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=duk1Q0f2DSzwZY75s%2Fbi4FM%2BMCVnv%2BmrBS7nl4IR2eY%3D&reserved=0> or Android<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cmtoniato%40alteanet.it%7Ca3c4262c9bb94c3c65bd08da079b956c%7Ca248a8b3ddc049da9a7070c77203e1d7%7C1%7C0%7C637830662050273821%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=XjxdpfGNdwRyHrZg6mlOndlUuRDJxXZQWo%2B40ojyZSE%3D&reserved=0>.

**Comment by alii-gator** [comment]
@mtoniato This is very unique concern. Usually the authids should be descriptive enough to identify the account name etc.
But I understand the concern, But this will require some more investigation because these credentials are not just used in VSCode but all the tools in SuiteCloud ecosystem (Suitecloud plugin for WebStorm, SDF CLI, SuiteCloud cli for NodeJS etc).

So we will wait for more cases on this and if we reach a sufficient amount of votes we can consider it for an enhancement.

**Comment by ShawnTalbert** [comment]
I agree the account list should be (at least optionally) a _project-specific_ list.
